pkgs/development/ruby-modules/gem/default.nix

```diff
@@ -142,6 +142,13 @@ stdenv.mkDerivation ((builtins.removeAttrs attrs ["source"]) // {
       gempkg=$(echo "$output" | grep -oP 'File: \K(.*)')
 
       echo "gem package built: $gempkg"
+    elif [[ "$type" == "git" ]]; then
+      ls .git | head -n 10
+      git init
+      git config user.email nixbld@localhost.none
+      git config user.name Nix
+      git add .
+      git commit -m "Nixpkgs setup for rubygems" || true
```

Comment:
I like this idea. Do you know if it's bit-reproducible?

You might be able to reduce a few lines by using `--author` and also add `--date=` just in case

Reply:
> I like this idea. Do you know if it's bit-reproducible?

It could definitely change between git versions. There’s a lot of stuff like initial repository config and sample hooks in a brand new repo.

Something like this could help:

https://github.com/NixOS/nixpkgs/blob/c51472854333a1236edf71d1972a0cc528c43704/pkgs/build-support/fetchgit/nix-prefetch-git#L245-L246

Aside from those, and possibly some other files like info/exclude, I think this would be bit-reproducible, unless the git repository format changed. There’s not really anything we could do about that, and it would already be a big problem for Nixpkgs.

Reply:
Excellent notes. I've removed the `commit` which removes the requirement for `config` and added that `rm`. Retesting on my problem repo still works.